katran/lib/KatranLb.cpp

```diff
@@ -1797,10 +1846,15 @@ void KatranLb::decreaseRefCountForReal(const folly::IPAddress& real) {
   }
 }
 
-uint32_t KatranLb::increaseRefCountForReal(const folly::IPAddress& real) {
+uint32_t KatranLb::increaseRefCountForReal(const folly::IPAddress& real, uint8_t flags) {
   auto real_iter = reals_.find(real);
+  flags &= ~V6DADDR; // to keep IPv4/IPv6 specific header
   if (real_iter != reals_.end()) {
     real_iter->second.refCount++;
+    reals_[real].flags = flags;
```

Comment:
this seems strange. why we override flags on already existing real? we have "modifyReal" for such cases